intersectionobserver/demo.js

```diff
@@ -13,16 +13,24 @@ function loadItems(n) {
 }
 
 var intersectionObserver = new IntersectionObserver(entries => {
-  // If intersectionRatio is 0, the sentinel is out of view
-  // and we do not need to do anything.
-  if (entries[0].intersectionRatio <= 0) {
-    return;
-  }
-  loadItems(10);
-  // appendChild will move the existing element, so there is no need to
-  // remove it first.
-  scroller.appendChild(sentinel);
-  loadItems(5);
-  ChromeSamples.setStatus('Loaded up to item ' + counter);
+  // If the browser is busy while scrolling happens, multiple entries can
+  // accumulate between invocations of this callback, so we loop over
+  // them.
+  entries.forEach(entry => {
+    // If intersectionRatio is 0, the sentinel is out of view
+    // and we do not need to do anything.
+    if (entry.intersectionRatio > 0) {
+      loadItems(10);
+      // appendChild will move the existing element, so there is no need to
+      // remove it first.
+      scroller.appendChild(sentinel);
+      loadItems(5);
+      ChromeSamples.setStatus('Loaded up to item ' + counter);
+
+      // We have already loaded more content, no need to look at any more
+      // entries.
+      return;
```

Comment:
This return only returns from the callback, which ends here anyway. The comment implies that the intent is to stop the `forEach` loop? Not quite sure

Reply:
Right, this 'return' statement ends the forEach loop.

Reply:
I feel like I’m missing something obvious. How is this `return` statement ending the `forEach`? You can’t break out of a `forEach`.

Reply:
Exactly. You would need to use `for...in` or a regular `for` loop to being able to break out. The problem with `forEach` is that the closure is executed for every element, and a `return` in the end of that closure has the same effect as having no such `return`.

Alternatively, you could also use `if(entries.findIndex(() => {...})) {...}

Reply:
+1 to `some()`

Reply:
*facepalm*

This is why I should never write javascript.

I switched to 'some', thanks for the suggestion.